src/core/pal/feature.cpp

```diff
@@ -1170,6 +1170,19 @@ int FeaturePart::createCurvedCandidatesAlongLine( QList< LabelPosition* >& lPos,
     return 0;
   }
 
+  //calculate overall angle of line
+  double lineAngle;
```

Comment:
is this block required? I removed it in https://github.com/qgis/QGIS/commit/ab782eb3faa95ff27731542041db7c250323b20c because it was throwing unused variable warnings, and I can't see where any of these variables are used.